Merging changes from GraalPy master and updating docs (#34)
graalvm/graalpy.github.io: docs/graalpy-embedded/reference/native/binary_size.md

## Reducing Binary Size

Python is a large language.
"Batteries included" has long been a core tenet of CPython.
As a compatible replacement, GraalPy includes most of those "batteries" as well.
This can result in significant increases in binary size when including GraalPy in a Java application.

Only you as the developer know your specific embedding scenario.
The defaults may include much more than needed for any specific application.
An embedded Python-in-Java application usually has more limited use cases for the Python interpreter than the full GraalPy distribution, and often you can know ahead of time whether certain features are needed.
Some features (for example, cryptographic algorithms or socket access) may even be undesirable to include in some cases.
Thus, when embedding GraalPy in a Java application, the binary size can be reduced and security improved by excluding components of the Python language.

### Excluding Python Components

> GraalPy defines a few system properties that can be passed on the `native-image` commandline to exclude aspects of the language.

Reviewer: ```suggestion
GraalPy defines a few system properties that can be passed on the `native-image` command line to exclude aspects of the language.
```

* `python.WithoutJNI=true` - This option removes any code that uses JNI. As a consequence, you cannot use the HPy JNI backend and maybe other parts that rely on JNI.

### Removing Pre-initialized Python Heap

Another useful option to reduce the size of the native executable is to omit a pre-initialized Python context from the executable.
By default, a default Python context is already pre-initialized and ready for immediate execution.
This can lead to slightly improved startup, at the cost of including a few thousand Python objects in the binary.
In embedded applications that use a custom polyglot engine to allow context sharing, the pre-initialized context cannot be used at all, and including those objects is wasted.
The pre-initialized heap can be omitted by passing the following to the `native-image` command:

```bash
-Dimage-build-time.PreinitializeContexts=
```

> ### Disabling Run-time Compilation of Python Code

Reviewer: ```suggestion
### Disabling Runtime Compilation of Python Code
```

> If binary size is significantly more important than execution speed (which may be the case if all Python scripts are expected to be short running, do a lot of IO, or scripts are rarely executed more than once), it may make sense to disable JIT compilation entirely.

Reviewer: ```suggestion
If binary size is significantly more important than execution speed (which may be the case if all Python scripts are expected to be short running, perform a lot of I/O, or scripts are rarely executed more than once), it may make sense to disable JIT compilation entirely.
```
